src/env.cpp

```diff
@@ -50,6 +57,9 @@ extern char **environ;
 
 bool g_use_posix_spawn = false;  // will usually be set to true
 
+/// Does the terminal have the "eat_newline_glitch".
+int term_has_xn = 0;
```

Comment:
Okay. It's only an int because that's what `tgetflag()` returns so it's six or a half-dozen which is clearer. I mostly want to avoid the legal and correct, but needlessly confusing, technique employed by zsh that puts a logical operator (`!hasxn`) in an arithmetic expression.